jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1455,7 +1455,9 @@ Method getMethodHelper(
 	boolean throwException, boolean forDeclaredMethod, List<Method> methodList, String name, Class<?>... parameterTypes)
 	throws NoSuchMethodException {
 	Method result;
+	Method resultFromRootSuperClass = null;
```

Comment:
It has to be initialized with null; otherwise it will end up with `error: variable resultFromRootSuperClass  might not have been initialized` in compilation.